### PR TITLE
Document the steps for creating a new `julia-release-1-dot-N` pipeline after a new `release-1.N` branch is made on JuliaLang/julia

### LINKED ISSUE #43: Documentation request: how to create the new `https://buildkite.com/julialang/julia-release-1-dot-NN/` pipeline after each feature freeze
Every time we have a new feature freeze, we need to remember to create the new `https://buildkite.com/julialang/julia-release-1-dot-NN/`.

E.g. we recently branched 1.8, so I had to create the `https://buildkite.com/julialang/julia-release-1-dot-8/` pipeline.

We need to document the steps for creating this new pipeline.

### PR DESCRIPTION
**Pull request by DilumAluthge**
Fixes #43

### CODE REVIEW

**Comment by DilumAluthge** [comment]
It sure would be nice if we could write a Julia script that takes in a Buildkite token and the value of `N` and uses the Buildkite API to perform all of the above steps automatically.

File: devdocs/create-new-release-pipeline.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - Step i: Make sure that the "Skip Intermediate Builds" checkbox IS checked.
+    - Step ii: Make sure that the "Cancel Intermediate Builds" checkbox IS checked.
+    - Step iii: Under "Skip Intermediate Builds", in the text box, enter the following text: `!master !main !release-*`
+    - Step iv: Under "Cancel Intermediate Builds", in the text box, enter the following text: `!master !main !release-*`
+    - Step v: Click on the green "Save Build Skipping" button.
+13. Go to https://buildkite.com/julialang/julia-release-1-dot-9 and click on the white "Pipeline Settings" button.
+14. In the left-hand side, under "Pipeline Settings", click on "GitHub". Then, in the "Branch Limiting" section, do the following actions:
+    - Step i: For "Branch Filter Pattern", enter the following text: `release-1.9 v1.9.*`
+    - Step ii: Click on the green "Save Branch Limiting" button.
+15. You are still on the "Pipeline Settings ⟶ GitHub" page. In the "GitHub Settings" section, do the following actions:
```
Comment (staticfloat) [inline devdocs/create-new-release-pipeline.md:32]: I'm thinking for things like this, perhaps a picture is easier?

**Comment by DilumAluthge** [comment]
Note to self: additional things that I need to add here:

https://buildkite.com/julialang/julia-release-1-dot-9/settings

—> General

—> Default Branch

Needs to be release-1.9

Also:
UNCHECK “Skip pull request builds for existing commits”

Also:
CHECK: cancel builds then the branch is deleted

File: devdocs/create-new-release-pipeline.md (hunk 1 above)
Comment (DilumAluthge) [reply devdocs/create-new-release-pipeline.md:32]: Yeah.

I'm going to merge this just to get something in (currently we have nothing on main), but we should make follow-up PRs to add some screenshots.

**Pull request merged**